### PR TITLE
Use a shared TableMenu

### PR DESCRIPTION
**Pull request by mikestone14**
This PR aims at front-end consistency by using consistent table menus and removing the padding from the active sidebar item which was causing a weird UI shift.

| Before | After |
| ------- | ----- |
| Screenshot 2023-11-27 at 11 07 21 AM | Screenshot 2023-11-27 at 11 04 11 AM |
| Screenshot 2023-11-27 at 11 07 33 AM | Screenshot 2023-11-27 at 11 04 22 AM |

### CODE REVIEW

File: views/tables/Table/TableMenu.jsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+const COLOR_STYLES = { color: 'black.A100' }
+const TableMenu = (props) => {
+  const { menuItems } = props
+  const [menuAnchor, setMenuAnchor] = React.useState(null)
+  const open = Boolean(menuAnchor)
+  const onClose = () => setMenuAnchor(null)
+
+  return (
+    <>
+      <Button
```
Comment (Elkrival) [inline views/tables/Table/TableMenu.jsx]: I wonder if we should use the IconButton for buttons that use Icons..?
https://mui.com/material-ui/api/icon-button/

File: views/tables/Table/TableMenu.jsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        id={`menu-${props.id}`}
+        anchorEl={menuAnchor}
+        open={open}
+        onClose={onClose}
+        MenuListProps={{
+          'aria-labelledby': `menu-button-${props.id}`,
+        }}
+      >
+        <MenuList sx={{ minWidth: '214px', maxWidth: '100%' }}>
+          {menuItems.map((menuItem) => (
```
Comment (Elkrival) [inline views/tables/Table/TableMenu.jsx]: maybe mui does this under the hood but the map is missing some keys

File: views/components/Sidebar/SidebarLink.css [hunk 3]
```diff
@@ -13,7 +13,6 @@
 .SidebarLink_link-active {
   display: flex;
   text-decoration: none;
-  padding: 6px;
   &:visited&:focus {
```
Comment (Elkrival) [inline views/components/Sidebar/SidebarLink.css:16]: I think we still need that padding or the sidebar highlight looks really thin

**Review by Elkrival — COMMENTED** [review]
Just a few things I noticed

File: views/components/Sidebar/SidebarLink.css (hunk 3 above)
Comment (mikestone14) [reply views/components/Sidebar/SidebarLink.css:16]: we still have padding, in fact we have more padding, by using the padding from the base, non-active class that is also on the active elements.

File: views/tables/Table/TableMenu.jsx (hunk 2 above)
Comment (mikestone14) [reply views/tables/Table/TableMenu.jsx]: it doesn't, good call - https://github.com/AMP-SCZ/dpdash/pull/494/commits/11517e6bfe6e3dc6e8a140beced61d2f18fbeb42

File: views/tables/Table/TableMenu.jsx (hunk 1 above)
Comment (mikestone14) [reply views/tables/Table/TableMenu.jsx]: yeah we can do that - https://github.com/AMP-SCZ/dpdash/pull/494/commits/11517e6bfe6e3dc6e8a140beced61d2f18fbeb42

**Review by Elkrival — APPROVED** [review]
🚢

**Pull request merged**